sdk/iot/pnp/inc/az_iot_pnp_client.h

```diff
@@ -0,0 +1,334 @@
+    az_span mqtt_topic_filter,
+    az_span* out_mqtt_topic_filter);
+
+/**
+ * @brief A command request received from the PnP service.
+ *
+ */
+typedef struct az_iot_pnp_client_command_request
+{
+  az_span request_id; /**< The request id.
+                       * @note The application must match the method request and method response. */
+  az_span component; /**< The component the command is targeted towards. */
+  az_span command_name; /**< The name of the command to invoke */
+  az_span payload; /**< The body of the request for the command, stripped of any envelope data from
+                        MQTT payload */
+} az_iot_pnp_client_command_request;
+
+/**
+ * @brief Attempts to parse a received message's topic.
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] received_topic An #az_span containing the received topic.
+ * @param[in] received_payload An #az_span containing the received payload.
+ * @param[out] out_request If the message is a command request, this will contain the
+ *                         #az_iot_pnp_client_command_request.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_command_received_topic_parse(
+    az_iot_pnp_client const* client,
+    az_span received_topic,
+    az_span received_payload,
+    az_iot_pnp_client_command_request* out_command_request);
+
+/**
+ * @brief Azure IoT PnP command response options.
+ *
+ */
+typedef struct az_iot_pnp_client_command_response_options
+{
+  struct
+  {
+    int _not_used;
+  } _internal;
+} az_iot_pnp_client_command_response_options;
+
+/**
+ * @brief Gets the default Azure IoT PnP command response options.
+ * @details Call this to obtain an initialized #az_iot_pnp_client_command_response_options structure
+ * that can be afterwards modified and passed to
+ * #az_iot_pnp_client_commands_response_publish_topic_get.
+ *
+ * @return #az_iot_pnp_client_command_response_options.
+ */
+AZ_NODISCARD az_iot_pnp_client_command_response_options az_iot_pnp_client_command_response_options_default();
+
+/**
+ * @brief Gets the MQTT topic that must be used to respond to command requests.
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] request_id The request id. Must match a received #az_iot_pnp_client_command_request
+ *                       request_id.
+ * @param[in] status The status. (E.g. 200 for success.)
+ * @param[in] options A reference to an #az_iot_pnp_client_command_response_options structure. Can
+ * be NULL.
+ * @param[in] mqtt_topic An empty #az_span with sufficient capacity to hold the MQTT topic.
+ * @param[out] out_mqtt_topic The output #az_span containing the MQTT topic.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_commands_response_publish_topic_get(
+    az_iot_pnp_client const* client,
+    az_span request_id,
+    uint16_t status,
+    az_iot_pnp_client_command_response_options const* options,
```

Comment:
This is he content type stuff, right?
If so, I think this should be on the client and not passed to each of these functions.

Reply:
No - the command_response options is currently not set to anything, and will be changed to a reserved per other discussions.